src/backend/utils/basic_analysis.js

```diff
@@ -71,18 +71,14 @@ module.exports = function analyzeText(text_) {
 
   // asynchonize function
   // to get a basic analysis information, getAsyAnalysis should be called
-  this.getAsyAnalysis = async function() {
-    return new Promise((res, rej) => {
-      if (isValidString) {
-        res(analysis);
-      } else {
-        res(textInfo);
-      }
-      if (isValidString === undefined) {
-        rej(textInfo);
-      }
-    });
+  this.getAsyAnalysis = function() {
+    if (isValidString === undefined) {
```

Comment:
It looks like this can all be simplified to:

```js
return isValidString ? analysis : textInfo;
```

Make sure I'm right (untested), but just reading this code, there's no reason for all these different code paths that I can see.